src/plugins/data/server/index_patterns/has_index_pattern_with_user_data.ts

```diff
@@ -0,0 +1,67 @@
+/*
```

Comment:
This is a slightly adjusted version from https://github.com/elastic/kibana/pull/108193,

I hoped we could delete code from `home` and use index pattern service there: https://github.com/elastic/kibana/issues/108711

But it isn't likely in case we change this implementation to search documents instead of using cat API: https://github.com/elastic/kibana/pull/108887#issuecomment-901032994